client/ayon_resolve/plugins/load/load_editorial_package.py

```diff
@@ -32,21 +36,126 @@ def load(self, context, name, namespace, data):
 
         project = lib.get_current_project()
         media_pool = project.GetMediaPool()
+        folder_path = context["folder"]["path"]
 
         # create versioned bin for editorial package
         version_name = context["version"]["name"]
-        bin_name = f"{name}_{version_name}"
-        lib.create_bin(bin_name)
-
+        loaded_bin = lib.create_bin(f"{folder_path}/{name}/{version_name}")
+        loaded_timeline_name = f"{name}_{version_name}_timeline"
         import_options = {
-            "timelineName": "Editorial Package Timeline",
+            "timelineName": loaded_timeline_name,
             "importSourceClips": True,
             "sourceClipsPath": search_folder_path.as_posix(),
         }
 
+        # import timeline from otio file
         timeline = media_pool.ImportTimelineFromFile(files, import_options)
+
+        # get timeline media pool item for metadata update
+        timeline_media_pool_item = None
+        for item in lib.iter_all_media_pool_clips(root=loaded_bin):
+            item_name = item.GetName()
+            if item_name != loaded_timeline_name:
+                continue
+            timeline_media_pool_item = item
+            break
```

Comment:
```suggestion
        timeline_media_pool_item = timeline.GetMediaPoolItem()
```

Reply:
**this** will result in:
```
Traceback (most recent call last):
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\tools\loader\models\actions.py", line 740, in _load_representations_by_loader
    load_with_repre_context(
  File "C:\CODE\__PYPE\ayon-core\client\ayon_core\pipeline\load\utils.py", line 325, in load_with_repre_context
    return loader.load(repre_context, name, namespace, options)
  File "C:\CODE\__PYPE\ayon-resolve\client\ayon_resolve\plugins\load\load_editorial_package.py", line 55, in load
    timeline_media_pool_item = timeline.GetMediaPoolItem()
TypeError: 'NoneType' object is not callable
```

Sadly *Timeline* object is having different set of attributes then *TimelineItem*.

Reply:
Ah of course! 

The tricky thing is that the name isn't entirely enforced to be unique unfortunately:
- https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=177766
- https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=178493

Both list the same issue. (Note that the name not being unique may also be due to a 'bug' but apparently there are ways to get resolve into that state)

Anyway - we might want to at least make this a dedicated lib function and ensure what we find is what we intend to find? How about this in `ayon_resolve.api.lib`:

```python
import uuid

def get_timeline_media_pool_item(timeline, root=None):
    """Return MediaPoolItem from Timeline"""
    
    name = timeline.GetName()
    
    # Due to limitations in the Resolve API we can't get
    # the media pool item directly from the timeline.
    # We can find it by name, however - but names are not
    # enforced to be unique across bins. So, we give it
    # unique name.
    identifier = str(uuid.uuid4().hex)
    try:
        timeline.SetName(identifier)
        for item in iter_all_media_pool_clips(root=root):
            item.GetName() == identifier:
                return item
        
    finally:
        # Revert to original name
        timeline.SetName(name)
        if item_name != loaded_timeline_name:
            continue
```

Reply:
thanks a lot for the suggestion, it is implemented